planner/core/rule_column_pruning.go

```diff
@@ -154,7 +154,7 @@ func (ls *LogicalSort) PruneColumns(parentUsedCols []*expression.Column) error {
 	for i := len(ls.ByItems) - 1; i >= 0; i-- {
 		cols := expression.ExtractColumns(ls.ByItems[i].Expr)
 		if len(cols) == 0 {
-			if !ls.ByItems[i].Expr.ConstItem() {
+			if expression.IsMutableEffectsExpr(ls.ByItems[i].Expr) {
```

Comment:
`CorCols` will be regarded as immutable by this function and then they will be pruned.
Is it OK to do that?

Reply:
It's expected behavior. Order by corcols is useless and can be pruned.

Reply:
OK, and please add some comments here about it.

Reply:
done!